docs/DeveloperGuide.md

```diff
@@ -2,37 +2,254 @@
+[comment]: <> (//@@author Jeraldchen)
+
+### User Interface
+![UIClass.png](diagrams%2FUIClass.png)
+
+This feature shows how the user interface is being implemented via three main classes which are Menu, Information and Parser.
+
+### Menu
+The menu class is responsible for the user interface of the application. It is responsible for displaying the menu and
+taking in user input. The menu class is also responsible for the main loop of the application, which is the loop that
+continues to run until the user exits the application. The menu class allows patients to log in, register and exit the
+program. 
+
+`showWelcomeMenu()` is used to display the welcome menu to the user.
+`register()` is used to register a new patient into the system.
+`login()` is used to log a patient into the system.
+`exit()` is used to exit the program.
+`showAccountMenu()` is used to display the list of choices to the user.
+`getUserSymptoms` is to ask input form user and store it in an arraylist.
+`addSymptomToSymptomList()` is used to add the symptom to the symptom list.
+`parseSymptomInput()` is used to parse the symptom input from the user.
+`addSymptoms()` is used to add the symptoms to the symptom list.
+`displaySymptomsList()` is used to display the list of symptoms to the user.
+`displayPossibleIllnesses()` is used to display the list of possible illnesses to the user.
+
+### Information
+The information class is responsible for displaying information to the user. It is responsible for displaying the
+information of the patient, the medicine and the diagnosis. It is also responsible for displaying the list of
+symptoms and the list of illnesses. A hashmap is used to store the information of the patient and the hashed password.
+
+`storePatientInfo()` is used to store the patient information into the hashmap. 
+`getPatientInfo()` is used to retrieve the patient information from the hashmap.
+`printDiagnosis()` is used to print the diagnosis of the patient.
+`resetDiagnosticHistory()` is used to reset and clear the diagnostic history of the patient.
+`checkHash()` is used to check if the hashed password is correct and matches.
+`getAllPatientData()` is used to retrieve all the patient data from the hashmap.
+`resetSymptomChoice()` is used to reset the symptom choice of the patient in case an error was made.
+`hashPassword()` is used to hash the password of the patient.
+
+### Parser
+The parser class is responsible for parsing the user input. It is responsible for parsing the user input from the
+menu and the symptom list. It is also responsible for parsing the user input from the medicine list.
+
+`parseWelcome()` is used to parse the user input from the welcome menu.
+`parseAccountCommand()` is used to parse the user command and then execute the command based on the choice number.
+
+[comment]: <> (//@@author Brennanzuz)
+### Diagnosis
+
+![DiagnosisClassDiagram.png](diagrams%2FDiagnosisClassDiagram.png)
```

Comment:
Maybe add more details for the aggregation as it is a little bit hard to understand.